carbonmark/components/pages/Projects/index.tsx

```diff
@@ -21,25 +25,61 @@ import { NextPage } from "next";
 import Link from "next/link";
 import { useRouter } from "next/router";
 import { ProjectsPageStaticProps } from "pages/projects";
+import { useEffect, useState } from "react";
+import { useForm, useWatch } from "react-hook-form";
 import { SWRConfig } from "swr";
+import { ProjectFilters } from "../Project/ProjectFilters";
+import { ProjectsController } from "../Project/ProjectsController";
 import * as styles from "./styles";
 
-const Page: NextPage = () => {
-  const { locale, query } = useRouter();
+type SortOption = keyof typeof PROJECT_SORT_OPTIONS;
 
-  const sortKey = String(query["sort"]);
+export type ModalFieldValues = {
+  sort: SortOption;
+  country: string[];
+  category: string[];
+  vintage: string[];
+};
 
-  const { projects, isLoading, isValidating } = useFetchProjects();
+export const DEFAULTS: ModalFieldValues = {
+  sort: "recently-updated",
+  country: [],
+  category: [],
+  vintage: [],
+};
 
-  const sortFn = get(PROJECT_SORT_FNS, sortKey) ?? identity;
+const Page: NextPage = () => {
+  const router = useRouter();
+  const [modalOpen, setModalOpen] = useState(false);
+  const defaultValues = { ...DEFAULTS, ...router.query };
```

Comment:
Similar to how I consume query params in the old app.klimadao.finance offset UI with a hook called [useOffsetParams](https://github.com/KlimaDAO/klimadao/blob/373c7deb32485bc88d530f6c12bba48e0330a8e2/app/lib/hooks/useOffsetParams.ts#L71), I would try to isolate all the query params logic into a hook or component if possible. (Don't copy that code as the requirements there are very different.)

The idea is the helper should consume, validate, and apply types to the query param values so the component can stay simple. This also avoids us passing unsupported params or typos into the form. Something like:

```ts
const { sort, countries } = useProjectsFilterParams();
const form = useForm({ sort, countries });
```

This hook should contain as much of the useEffect logic as you can as well.

Reply:
Sure, will look at tidying this up and moving to a dedicated hook or component.